### PR TITLE
Feature/sec 17 1

### PR DESCRIPTION
**Pull request by gizet**

### CODE REVIEW

File: src/main/java/ro/secur/skillsdata/configuration/JwtConfig.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+@Component
+@ConfigurationProperties("application.jwt")
+public class JwtConfig {
+
+    private String secretKey;
+
+    private String tokenPrefix;
+
+    private Integer ExpirationDays;
```
Comment (CosminaIacob) [inline src/main/java/ro/secur/skillsdata/configuration/JwtConfig.java]: camel case please

Comment (CosminaIacob) [inline src/main/java/ro/secur/skillsdata/configuration/JwtConfig.java]: first letter lowercase please

Comment (gizet) [reply src/main/java/ro/secur/skillsdata/configuration/JwtConfig.java]: copy paste

Comment (gizet) [reply src/main/java/ro/secur/skillsdata/configuration/JwtConfig.java]: done

File: src/main/java/ro/secur/skillsdata/security/filter/TokenVerifier.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+        String token = authorizationHeader.replace("Bearer ", "");
+        try {
+            Jws<Claims> claimsJws = Jwts.parser()
+                    .setSigningKey(jwtConfiguration.secretKey())
+                    .parseClaimsJws(token);
+
+            Claims body = claimsJws.getBody();
+            String username = body.getSubject();
+            var authorities = (List<Map<Strings, String>>) body.get("roles");
```
[the remark this answers is not in the document]
Comment (gizet) [reply src/main/java/ro/secur/skillsdata/security/filter/TokenVerifier.java]: Used a static class till we find a better solution

File: src/main/java/ro/secur/skillsdata/security/filter/TokenVerifier.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+            Set<SimpleGrantedAuthority> simpleGrantedAuthority = authorities.stream()
+                    .map(m -> new SimpleGrantedAuthority(m.get("authority")))
```
[the remark this answers is not in the document]
Comment (gizet) [reply src/main/java/ro/secur/skillsdata/security/filter/TokenVerifier.java]: done

File: src/main/java/ro/secur/skillsdata/security/filter/TokenVerifier.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+            Claims body = claimsJws.getBody();
+            String username = body.getSubject();
+            var authorities = (List<Map<Strings, String>>) body.get(ROLES);
+            Set<SimpleGrantedAuthority> simpleGrantedAuthority = authorities.stream()
+                    .map(m -> new SimpleGrantedAuthority(m.get(AUTHORITY)))
+                    .collect(Collectors.toSet());
+
+
+            Authentication authentication = new PreAuthenticatedAuthenticationToken(
```
Comment (ftansanu) [inline src/main/java/ro/secur/skillsdata/security/filter/TokenVerifier.java]: refactor lines 50-62 in the same way as we have in auth

Comment (gizet) [reply src/main/java/ro/secur/skillsdata/security/filter/TokenVerifier.java]: ok

**Pull request merged**